src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/PasteOrdering.cs

```diff
@@ -0,0 +1,137 @@
+        private readonly IProjectAccessor _accessor;
+
+        private IProjectTree _dropTarget;
+
+        private IPasteHandler _pasteHandler;
+        private IPasteDataObjectProcessor _pasteProcessor;
+
+        [ImportingConstructor]
+        public PasteOrdering(UnconfiguredProject unconfiguredProject, IProjectAccessor accessor)
+        {
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+            Requires.NotNull(accessor, nameof(accessor));
+
+            _configuredProject = unconfiguredProject.Services.ActiveConfiguredProjectProvider.ActiveConfiguredProject;
+            _accessor = accessor;
+
+            PasteHandlers = new OrderPrecedenceImportCollection<IPasteHandler>(projectCapabilityCheckProvider: unconfiguredProject);
+            PasteProcessors = new OrderPrecedenceImportCollection<IPasteDataObjectProcessor>(projectCapabilityCheckProvider: unconfiguredProject);
+        }
+
+        [ImportMany]
+        private OrderPrecedenceImportCollection<IPasteHandler> PasteHandlers { get; set; }
+
+        [ImportMany]
+        private OrderPrecedenceImportCollection<IPasteDataObjectProcessor> PasteProcessors { get; set; }
+
+        private IPasteHandler PasteHandler
+        {
+            get
+            {
+                if (_pasteHandler == null)
+                {
+                    // Grab the paste handler that has the highest order precedence that is below PasteOrdering's order precedence. 
+                    _pasteHandler =
+                        PasteHandlers.Where(x => x.Metadata.OrderPrecedence < OrderPrecedence)
+                        .OrderByDescending(x => x.Metadata.OrderPrecedence).First().Value;
+                }
+
+                Assumes.NotNull(_pasteHandler);
+
+                return _pasteHandler;
+            }
+        }
+
+        private IPasteDataObjectProcessor PasteProcessor
+        {
+            get
+            {
+                if (_pasteProcessor == null)
+                {
+                    // Grab the paste processor that has the highest order precedence that is below PasteOrdering's order precedence. 
+                    _pasteProcessor =
+                        PasteProcessors.Where(x => x.Metadata.OrderPrecedence < OrderPrecedence)
+                        .OrderByDescending(x => x.Metadata.OrderPrecedence).First().Value;
+                }
+
+                Assumes.NotNull(_pasteProcessor);
+
+                return _pasteProcessor;
+            }
+        }
+
+        public bool CanHandleDataObject(object dataObject, IProjectTree dropTarget, IProjectTreeProvider currentProvider)
+        {
+            return PasteProcessor.CanHandleDataObject(dataObject, dropTarget, currentProvider);
+        }
+
+        public Task<IEnumerable<ICopyPasteItem>> ProcessDataObjectAsync(object dataObject, IProjectTree dropTarget, IProjectTreeProvider currentProvider, DropEffects effect)
+        {
+            _dropTarget = dropTarget;
```

Comment:
Help me understand why you store this?

Reply:
When you drop and paste items into a folder, this is that target folder. I need to capture the target in order to perform a move in PasteItemsAsync.